test/integration/job/job_test.go

```diff
@@ -1026,6 +1026,334 @@ func TestIndexedJob(t *testing.T) {
 	validateTerminatedPodsTrackingFinalizerMetric(t, 5)
 }
 
+// This test has 4 parts which execute in the following order:
+// 1) Create Indexed Job then succeed/fail some indexes
+// 2) Optional: scale up then succeed/fail some indexes
+// 3) Optional: scale down then succeed/fail some indexes
+// 4) Optional: scale back up again
+//
+// Parts 1-3 are only executed if a flag in the test case
+// indicates that they should do so. Therefore, each test
+// case can toggle them to test various scenarios involving
+// scaling up/down/scaling up again, etc.
+func TestElasticIndexedJob(t *testing.T) {
```

Comment:
I feel it's better to split this test into multiple tests rather than trying to fit everything into table format.

Reply:
I think we can still have a table and potentially simplify things by creating a struct that describes a single update and the expectations:

```
type update struct {
  completions int
  succeededIndexes []int
  failedIndexes []int
  wantSucceeded int
  wantSucceededIndexes string
  wantFailed int
} 
```

then the test cases can be described as a list of those update:

```
	testCases := []struct {
		featureGate bool
		updates      []update
        }
```


@danielvegamyhre is this doable?

Reply:
Great idea! I will refactor this and let you know when it's done.

Reply:
I rewrote the test using this approach and all cases are passing.